src/test/scala/mesosphere/marathon/upgrade/ScalingCalculatorTest.scala

```diff
@@ -0,0 +1,82 @@
+package mesosphere.marathon.upgrade
+
+import mesosphere.marathon.Protos.MarathonTask
+import mesosphere.marathon.state.Timestamp
+import mesosphere.marathon.tasks.MarathonTasks
+import org.scalatest.{ Matchers, FunSuite }
+
+class ScalingCalculatorTest extends FunSuite with Matchers {
```

Comment:
I really like that you factored out the logic of which tasks to kill. The tests are easy to understand.